### PR TITLE
Apply API rate limits by user name instead of ip address

### PR DESCRIPTION
**Pull request by iamigo**
Limit by user name (instead of ip address).
Add tests.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -214,6 +205,19 @@ function start() { // eslint-disable-line max-statements
...
 
+    /*
+     * Set up API rate limits. Note that we are doing this *after* the
+     * swaggerSecurity middleware so that jwtUtil.verifyToken will already
+     * have been executed.
+     */
+    limiter({
+      path: conf.endpointToLimit,
+      method: conf.httpMethodToLimit,
+      lookup: ['headers.UserName'], // use 'x-forwarded-for' for ip address
```
Comment (shriramshankar) [inline index.js]: Looks like the comment needs to be updated or removed

File: tests/limiter/limiter.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    })
+    .then((u2) => tu.createTokenFromUserName(u2.name))
+    .then((tok2) => {
+      token2 = tok2;
+      done();
+    })
+    .catch(done);
+  });
+  after(u.forceDelete);
+  after(tu.forceDeleteUser);
```
Comment (pallavi2209) [inline tests/limiter/limiter.js]: Force delete token as well?

**Comment by iamigo** [comment]
Updated per comments @shriramshankar @pallavi2209

File: tests/limiter/limiter.js (hunk 2 above)
Comment (iamigo) [reply tests/limiter/limiter.js]: done

**Pull request merged**